hydrosdk/predictor.py

```diff
@@ -1,72 +1,133 @@
+        :param inputs: dict in the format of {string: Union[python_primitive_types, numpy_primitive_type]} with contract info
+        :param model_spec: model specification created for associated servable
+        :return: PredictResponse with outputs in protobuf format
+        """
+        # TODO: add проверка inputs на соответсиве сигнатуре
+        inputs_as_proto = convert_inputs_to_tensor_proto(inputs, self.signature)
+
+        try:
+            response = self.impl.send_data(model_spec=self.model_spec, inputs=inputs_as_proto)
+
+            if self.return_type == PredictorDT.DF:
+                return self.predict_resp_to_df(response=response)
+
+            elif self.return_type == PredictorDT.DICT_NP_ARRAY:
+                return self.predict_resp_to_dict_nparray(response=response)
+
+            elif self.return_type == PredictorDT.DICT_PYTHON:
+                return self.predict_resp_to_dict_pydtype(response=response)
+        except Exception as err:
+            raise Exception(f"Failed to predict.{str(err)} ")
+
+    @staticmethod
+    def predict_resp_to_dict_pydtype(response: PredictResponse) -> dict:
+        output_tensors_dict = {}
+        for tensor_name, tensor_proto in response.outputs.items():
+            dims = [dim.size for dim in tensor_proto.tensor_shape.dim]
+            value = getattr(tensor_proto, DTYPE_TO_FIELDNAME[tensor_proto.dtype])
+
+            # If no dims specified in TensorShapeProto, then it is scalar
+            if dims:
+                value = np.reshape(value, dims).tolist()
+                output_tensors_dict[tensor_name] = value
+            else:
+                output_tensors_dict[tensor_name] = value[0]
+
+        return output_tensors_dict
+
+    @staticmethod
+    def predict_resp_to_dict_nparray(response: PredictResponse) -> dict:
+        output_tensors_dict = {}
+        for tensor_name, tensor_proto in response.outputs.items():
+            array_shape = [dim.size for dim in tensor_proto.tensor_shape.dim]
+            np_dtype = proto2np_dtype(tensor_proto.dtype)
+            value = getattr(tensor_proto, DTYPE_TO_FIELDNAME[tensor_proto.dtype])
+            np_array_value = np.array(value, dtype=np_dtype)
+
+            # If no dims specified in TensorShapeProto, then it is scalar
+            if array_shape:
+                output_tensors_dict[tensor_name] = np_array_value.reshape(*array_shape)
+            else:
+                output_tensors_dict[tensor_name] = np.asscalar(np_array_value)
+
+        return output_tensors_dict
+
+    @staticmethod
+    def predict_resp_to_df(response: PredictResponse) -> pd.DataFrame:
+        response_dict: Dict[str, np.array] = PredictServiceClient.predict_resp_to_dict_nparray(response)
+        return pd.DataFrame(response_dict)
+
+
+class Predictable:
+    """Adds Predictor functionality"""
+    def predictor(self, monitorable=False, return_type=PredictorDT.DICT_NP_ARRAY) -> PredictServiceClient:
```

Comment:
`monitorable` should be `True` by default

Reply:
Fixed